overwatch/src/overwatch/handle.rs

```diff
@@ -120,25 +121,112 @@ where
         })
     }
 
-    /// Send a shutdown signal to the
-    /// [`OverwatchRunner`](crate::overwatch::OverwatchRunner)
-    pub async fn shutdown(&self) {
-        info!("Shutting down Overwatch");
+    /// Send a start signal to the specified service.
+    ///
+    /// # Errors
+    ///
+    /// If the start signal cannot be successfully delivered to the specified
+    /// service.
+    pub async fn start_service<Service>(&self) -> Result<(), ServiceError>
+    where
+        RuntimeServiceId: AsServiceId<Service>,
+    {
+        info!("Starting service with ID {}", RuntimeServiceId::SERVICE_ID);
+
+        let (sender, mut receiver) = tokio::sync::broadcast::channel(1);
+        self.send(OverwatchCommand::ServiceLifeCycle(
+            ServiceLifeCycleCommand {
+                service_id: RuntimeServiceId::SERVICE_ID,
+                msg: LifecycleMessage::Start(sender),
+            },
+        ))
+        .await
+        .map_err(|e| {
+            dbg!(e);
+            ServiceError::Start
+        })?;
+        receiver.recv().await.map_err(|e| {
+            dbg!(e);
+            ServiceError::Start
+        })?;
+        Ok(())
+    }
+
+    /// Send a start signal to the
+    /// [`OverwatchRunner`](crate::overwatch::OverwatchRunner) signaling it
+    /// to start all services.
+    ///
+    /// # Errors
+    ///
+    /// Fails silently if the start signal cannot be sent.
+    pub async fn start_all_services(&self) {
+        info!("Starting all services");
         let _: Result<(), _> = self
             .send(OverwatchCommand::OverwatchLifeCycle(
-                OverwatchLifeCycleCommand::Shutdown,
+                OverwatchLifeCycleCommand::StartAllServices,
+            ))
+            .await
+            .map_err(|e| dbg!(e));
+    }
+
+    /// Send a stop signal to the specified service.
+    ///
+    /// # Errors
+    ///
+    /// If the stop signal cannot be successfully delivered to the specified
+    /// service.
+    pub async fn stop_service<Service>(&self) -> Result<(), ServiceError>
+    where
+        RuntimeServiceId: AsServiceId<Service>,
+    {
+        info!("Stopping service with ID {}", RuntimeServiceId::SERVICE_ID);
+
+        let (sender, mut receiver) = tokio::sync::broadcast::channel(1);
+        self.send(OverwatchCommand::ServiceLifeCycle(
+            ServiceLifeCycleCommand {
+                service_id: RuntimeServiceId::SERVICE_ID,
+                msg: LifecycleMessage::Stop(sender),
+            },
+        ))
+        .await
+        .map_err(|e| {
+            dbg!(e);
+            ServiceError::Stop
+        })?;
+        receiver.recv().await.map_err(|e| {
```

Comment:
Same comment as above. We don't seem to use the provided `sender` for handling stop messages either.